libs/auth-api-lib/src/lib/delegations/delegations-incoming-custom.service.ts

```diff
@@ -156,21 +186,44 @@ export class DelegationsIncomingCustomService {
     return { delegations: aliveDelegations, fromNameInfo }
   }
 
-  private checkIfScopeAllowed(
+  private checkIfScopeIsValid(
     scope: DelegationScope,
-    allowedScopes: string[],
+    customApiScopes: ApiScope[],
   ): boolean {
-    return allowedScopes.includes(scope.scopeName)
+    return customApiScopes.map((s) => s.name).includes(scope.scopeName)
   }
 
-  private async getClientAllowedScopes(user: User) {
-    return (
-      await this.clientAllowedScopeModel.findAll({
-        where: {
-          clientId: user.client,
-        },
-      })
-    ).map((s) => s.scopeName)
+  private checkIfScopeAllowed(
+    scope: DelegationScope,
+    customApiScopes: ApiScope[],
+    accesses: ApiScopeUserAccess[],
+    fromNationalId: string,
+  ): boolean {
+    if (
+      customApiScopes
+        .filter((s) => !s.isAccessControlled)
+        .map((s) => s.name)
+        .includes(scope.scopeName)
```

Comment:
This function runs in a loop so would be nice to extra array allocation. Can use `customApiScopes.find()` here.

Reply:
Fixed.